File: mcp_server_development/node/src/review02/q4-server.ts
/**
 * 社内お知らせ掲示板サーバー ― stdio トランスポートのエントリーポイント
 *
 * 実行： docker compose exec node npx tsx src/review02/q4-server.ts
 * （単体で起動すると stderr に 1 行出したあと沈黙します。それが正常です）
 *
 * stdout は JSON-RPC の通信路そのものなので console.log は絶対に使いません。
 */
import { StdioServerTransport } from "@modelcontextprotocol/sdk/server/stdio.js";

import { createNoticeServer } from "./q4-create-server.js";

const server = createNoticeServer();
await server.connect(new StdioServerTransport());

console.error("[notice-board] stdio でリクエストを待機しています");

File: mcp_server_development/python/src/review01/q4_server.py
"""日次ダッシュボードサーバー（Python 版）― stdio トランスポートのエントリーポイント

このファイルの仕事は「サーバー定義を stdio につなぐ」ことだけです。

重要：stdio では標準出力（stdout）が JSON-RPC の通信路そのものです。
print() を 1 回でも呼ぶと電文が壊れます。ログは必ず stderr へ。

  docker compose exec python python src/review01/q4_server.py
"""

import sys

from q4_create_server import mcp

if __name__ == "__main__":
    # mcp.run() はここでブロックして戻ってこないので、ログは run の前に出す
    print("[team-dashboard-daily] stdio でリクエストを待機しています", file=sys.stderr)
    mcp.run(transport="stdio")
